crates/runtime/src/init/dataset.rs

```diff
@@ -604,11 +605,16 @@ impl Runtime {
             if let Some(current_ds) = existing_datasets.iter().find(|d| d.name == ds.name) {
                 if ds != *current_ds {
                     self.update_dataset(ds).await;
+                } else if ds.is_file_accelerated() {
+                    let datasets = self.initialize_accelerators(&[Arc::clone(&ds)]).await;
```

Comment:
Can't we call `initialize_accelerators for all `valid_datasets` before iterating over them one-by-one?